src/actions/src.cr

```diff
@@ -1,19 +1,96 @@
+require "dppm/prefix/src"
+
 module DppmRestApi::Actions::Src
   extend self
   include RouteHelpers
 
+  struct MultiSet(K, V)
+    @data = {} of K => Set(V)
+
+    def add(at k : K, v : V) : Nil
+      @data[k] = Set{v} unless @data[k]?.try { |set| set << v }
+    end
+
+    def add(at k : K, v : Enumerable(V))
+      if @data[k]?
+        @data[k] &= v.to_set
+      else
+        @data[k] = Set(V).new v
+      end
+    end
+
+    forward_missing_to @data
+  end
+
+  struct ::DPPM::Prefix::PkgFile
+    # Yield each instance variable to the block
+    def each_ivar
+      {% for ivar in @type.instance_vars %}
+        yield {{ ivar.stringify }}, @{{ivar.id}}
+      {% end %}
+    end
+
+    # Build a JSON response of the config keys mapped to their values, but only
+    # if they're specified in the *only* list. If *only* is empty, all values
+    # are forwarded
+    def to_json(builder : JSON::Builder, only : Enumerable(String))
```

Comment:
`only` isn't self-explanatory enough.
I suggest to use a block for filtering, like `Enumerable#select`

Reply:
I don't like the idea of using a block. It would either be less clear, in the case of a syntax like...
```crystal
src.pkg_file.to_json response do |key|
```
...because it's not super clear what it is we're *do*ing with _key_. In the case of a syntax like...
```crystal
src.pkg_file.select { |key| ... }.to_json response
```
...it would require building of an intermediate data structure, and even if it's just `Iterator.of`, that has some significant overhead associated with it over the simple `each_ivar` macro.

That said I agree the wording could be more clear. I thought I had seen `only`, and [`except`](https://github.com/DFabric/dppm-rest-api/blob/2defa42b63eaa5ee6e6c4a6dcb75b0e3e0400106/src/config/user.cr#L80) somewhere, but I can't recall where, and there's no reason _seeing it before_ makes something gospel haha. What about `expose` and `redact`?

Reply:
Replacing `only` with `keys` would be enough to clarify